numpy/core/arrayprint.py

```diff
@@ -485,7 +485,7 @@ def _array2string(a, options, separator=' ', prefix=""):
     # skip over array(
     next_line_prefix += " "*len(prefix)
 
-    lst = _formatArray(a, format_function, options['linewidth'],
+    lst = _formatArray(asarray(a), format_function, options['linewidth'],
```

Comment:
Just use `data`? And add a comment by the `asarray` there about recursion too.

Reply:
`data` doesn't work because it was passed through `_leading_trailing`. I'll add a comment.